.github/workflows/nightly-builds.yml

```diff
@@ -0,0 +1,141 @@
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+
+      - name: Set up JDK 8
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.8.0
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.2
+
+      - name: sbt akka-cluster-metrics/test
+        run: |-
+          sbt -jvm-opts .jvmopts-ci \
+            -Djava.security.egd=file:/dev/./urandom \
+            -Dakka.test.sigar=true \
+            -Dakka.cluster.assert=on \
+            -Dakka.test.timefactor=10 \
+            -Dakka.test.tags.exclude=gh-exclude \
+            clean akka-cluster-metrics/test
+
+      - name: Test Reports
+        # Makes it easier to spot failures instead of looking at the logs.
+        if: ${{ failure() }}
+        uses: marcospereira/action-surefire-report@v1
+        with:
+          report_paths: '**/target/test-reports/TEST-*.xml'
+          fail_if_no_tests: false
+          skip_publishing: true
+
+      - name: Email on failure
+        if: ${{ failure() }}
+        uses: dawidd6/action-send-mail@v3
+        with:
+          server_address: smtp.gmail.com
+          server_port: 465
+          username: ${{secrets.MAIL_USERNAME}}
+          password: ${{secrets.MAIL_PASSWORD}}
+          subject: "Failed: ${{ github.workflow }} / ${{ github.job }}"
+          to: akka.official@gmail.com
+          from: Akka CI (GHActions)
+          body: |
+            Job ${{ github.job }} in workflow ${{ github.workflow }} of ${{github.repository}} failed!
+            https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}
+
+  jdk-nightly-tests:
+    name: JDK ${{ matrix.jdkVersion }} / Scala ${{ matrix.scalaVersion }}
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        # No need to specify the full Scala version. Only the Scala binary version
+        # is required and Akka build will set the right full version from it.
+        scalaVersion: ["2.12", "2.13"]
+        jdkVersion: ["adopt@1.8.0", "adopt@1.11"]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK ${{ matrix.jdkVersion }}
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: ${{ matrix.jdkVersion }}
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.2
+
+      # Only run multijvm tests in Scala 2.13. No need to run it to
+      # multiple Scala versions AND multiple JDK versions. This will
+      # run multijvm tests to run for both JDK 8 and 11 though.
+      - name: Compile and Test
+        run: |-
+          ## TODO: restore the multi-node tests
+          sbt -jvm-opts .jvmopts-ci \
+            -Dakka.build.scalaVersion=${{ matrix.scalaVersion }} \
+            -Dakka.test.timefactor=10 \
```

Comment:
We have tried to make tests handle rather random pauses of 2-3 seconds on Jenkins. Default timeout for asserts is 3 seconds. Multiplying that with 10 makes no sense to me. Pauses of 30 seconds from the CI environment is not something we can deal with, and I don't gh actions will have such delays.

I can understand if we want to try with `timefactor=2` but if things are still flaky with that we have other hardening work to do.

Reply:
> I can understand if we want to try with `timefactor=2` but if things are still flaky with that we have other hardening work to do.

We can go with 2, I agree. I was using the same value we use now for Lagom.

Reply:
Done in 35dbab7b582d26d4d4cb36675bb1d93186bd5f03.